tests/ssl-opt.sh

```diff
@@ -5221,6 +5221,18 @@ run_test    "DHM size: server default, client 2049, rejected" \
             1 \
             -c "DHM prime too short:"
 
+# Check that the SM2 curve is not used for DHM
+run_test    "DHM: no SM2" \
```

Comment:
Minor: by DHM we usually mean finite-field Diffie-Hellman(-Merkle), please use ECDH here and in the comment above.